### ISSUE TITLE
Error when fetching balance from Bithumb

### ISSUE THREAD

**Issue by carloconnor** [ccxt/ccxt#6704]
- OS: Windows 10 Pro
- Programming Language version: Python 3.7.3
- CCXT version: ccxt==1.25.41

```
import ccxt
exchange = ccxt.bithumb({'enableRateLimit': True, "timeout": 10000, "apiKey": "a", "secret": "s"})
balance = exchange.fetch_balance()
```

The verbose shows the request succeeds:
Request: GET https://api.bithumb.com/public/ticker/all {'User-Agent': 'python-requests/2.22.0', 'Accept-Encoding': 'gzip, deflate'} None

Response: GET https://api.bithumb.com/public/ticker/all 200 [Very long response]

Request: POST https://api.bithumb.com/info/balance {'Accept': 'application/json', 'Content-Type': 'application/x-www-form-urlencoded', 'Api-Key': 'a', 'Api-Sign': 'my_signature', 'Api-Nonce': '1585583708934', 'User-Agent': 'python-requests/2.22.0', 'Accept-Encoding': 'gzip, deflate'} endpoint=%2Finfo%2Fbalance&currency=ALL

Response: POST https://api.bithumb.com/info/balance 200 {'Date': 'Mon, 30 Mar 2020 15:59:24 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Set-Cookie': '__cfduid=cookie_value; expires=Wed, 29-Apr-20 15:59:23 GMT; path=/; domain=.bithumb.com; HttpOnly; SameSite=Lax, WMONID=some_id; expires=Tue, 30-Mar-2021 15:59:24 GMT; Max-Age=31536000; path=/', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': '*', 'Access-Control-Allow-Headers': '*', 'CF-Cache-Status': 'DYNAMIC', 'Expect-CT': 'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"', 'Server': 'cloudflare', 'CF-RAY': '57c3099eaf60ca63-YUL', 'Content-Encoding': 'gzip'} {"status":"0000","data":[I see my good balance info here]}

Traceback (most recent call last):
  File "C:\...\site-packages\ccxt\bithumb.py", line 155, in fetch_balance
    account['total'] = self.safe_float(balances, 'total_' + lowerCurrencyId)
TypeError: must be str, not NoneType

If I go back to the previous ccxt version I had (1.21.32), it works well. I notice that the code in the method fetch_balance of bithumb.py has changed between both versions, the old one did:
```
currencyId = currency['id']
lowercase = currencyId.lower()
```

And the new code does:
```
lowerCurrencyId = self.safe_string_lower(currency, 'id')
```

Which results in lowerCurrencyId being None.

Thanks!

**Comment by kroitor** [comment]
Can you plz paste the full verbose response? Need to see which key is breaking the parser there.

**Comment by carloconnor** [comment]
Here's the balance info that I truncated in my initial post.

{"total_krw":"63449.880000","in_use_krw":"0.00000000","available_krw":"63449.88000000","total_btc":"0.00000000","in_use_btc":"0.00000000","available_btc":"0.00000000","xcoin_last_btc":"7740000","total_eth":"0.274013956190476190","in_use_eth":"0.00000000","available_eth":"0.27401395","xcoin_last_eth":"161000","total_dash":"0.00000000","in_use_dash":"0.00000000","available_dash":"0.00000000","xcoin_last_dash":"79150","total_ltc":"0.00000000","in_use_ltc":"0.00000000","available_ltc":"0.00000000","xcoin_last_ltc":"47280","total_etc":"0.000000000000000000","in_use_etc":"0.00000000","available_etc":"0.00000000","xcoin_last_etc":"6000","total_xrp":"0.00000000","in_use_xrp":"0.00000000","available_xrp":"0.00000000","xcoin_last_xrp":"210.1","total_bch":"0.00000000","in_use_bch":"0.00000000","available_bch":"0.00000000","xcoin_last_bch":"265500","total_xmr":"0.000000000000","in_use_xmr":"0.00000000","available_xmr":"0.00000000","xcoin_last_xmr":"57400","total_zec":"0.00000000","in_use_zec":"0.00000000","available_zec":"0.00000000","xcoin_last_zec":"37040","total_qtum":"0.00000000","in_use_qtum":"0.00000000","available_qtum":"0.00000000","xcoin_last_qtum":"1465","total_btg":"0.00000000","in_use_btg":"0.00000000","available_btg":"0.00000000","xcoin_last_btg":"8655","total_eos":"0.00000000","in_use_eos":"0.00000000","available_eos":"0.00000000","xcoin_last_eos":"2692","total_icx":"0.000000000000000000","in_use_icx":"0.00000000","available_icx":"0.00000000","xcoin_last_icx":"248.2","total_ven":"0.000000000000000000","in_use_ven":"0.00000000","available_ven":"0.00000000","xcoin_last_ven":"0","total_trx":"0.00000000","in_use_trx":"0.00000000","available_trx":"0.00000000","xcoin_last_trx":"14.17","total_elf":"0.000000000000000000","in_use_elf":"0.00000000","available_elf":"0.00000000","xcoin_last_elf":"72.41","total_mith":"0.000000000000000000","in_use_mith":"0.00000000","available_mith":"0.00000000","xcoin_last_mith":"0","total_mco":"0.00000000","in_use_mco":"0.00000000","available_mco":"0.00000000","xcoin_last_mco":"5695","total_omg":"0.000000000000000000","in_use_omg":"0.00000000","available_omg":"0.00000000","xcoin_last_omg":"619.8","total_knc":"0.000000000000000000","in_use_knc":"0.00000000","available_knc":"0.00000000","xcoin_last_knc":"535.8","total_gnt":"0.000000000000000000","in_use_gnt":"0.00000000","available_gnt":"0.00000000","xcoin_last_gnt":"40.75","total_hsr":"0.00000000","in_use_hsr":"0.00000000","available_hsr":"0.00000000","xcoin_last_hsr":"0","total_zil":"0.000000000000","in_use_zil":"0.00000000","available_zil":"0.00000000","xcoin_last_zil":"4.601","total_ethos":"0.00000000","in_use_ethos":"0.00000000","available_ethos":"0.00000000","xcoin_last_ethos":"0","total_pay":"0.000000000000000000","in_use_pay":"0.00000000","available_pay":"0.00000000","xcoin_last_pay":"0","total_waxp":"0.00000000","in_use_waxp":"0.00000000","available_waxp":"0.00000000","xcoin_last_waxp":"35.67","total_powr":"0.00000000","in_use_powr":"0.00000000","available_powr":"0.00000000","xcoin_last_powr":"63.47","total_lrc":"0.000000000000000000","in_use_lrc":"0.00000000","available_lrc":"0.00000000","xcoin_last_lrc":"32.18","total_gto":"0.00000000","in_use_gto":"0.00000000","available_gto":"0.00000000","xcoin_last_gto":"0","total_pch":"0.000000000000000000","in_use_pch":"0.00000000","available_pch":"0.00000000","xcoin_last_pch":"0","total_eosdac":"0.00000000","in_use_eosdac":"0.00000000","available_eosdac":"0.00000000","xcoin_last_eosdac":"0","total_steem":"0.00000000","in_use_steem":"0.00000000","available_steem":"0.00000000","xcoin_last_steem":"173.5","total_strat":"0.00000000","in_use_strat":"0.00000000","available_strat":"0.00000000","xcoin_last_strat":"290.2","total_ae":"0.000000000000000000","in_use_ae":"0.00000000","available_ae":"0.00000000","xcoin_last_ae":"117.6","total_zrx":"0.000000000000000000","in_use_zrx":"0.00000000","available_zrx":"0.00000000","xcoin_last_zrx":"184.3","total_rep":"0.000000000000000000","in_use_rep":"0.00000000","available_rep":"0.00000000","xcoin_last_rep":"12140","total_xem":"0.00000000","in_use_xem":"0.00000000","available_xem":"0.00000000","xcoin_last_xem":"44.13","total_snt":"0.000000000000000000","in_use_snt":"0.00000000","available_snt":"0.00000000","xcoin_last_snt":"22.01","total_ada":"0.00000000","in_use_ada":"0.00000000","available_ada":"0.00000000","xcoin_last_ada":"36.22","total_ppt":"0.00000000","in_use_ppt":"0.00000000","available_ppt":"0.00000000","xcoin_last_ppt":"0","total_ctxc":"0.000000000000000000","in_use_ctxc":"0.00000000","available_ctxc":"0.00000000","xcoin_last_ctxc":"94.24","total_bat":"0.000000000000000000","in_use_bat":"0.00000000","available_bat":"0.00000000","xcoin_last_bat":"171","total_wtc":"0.000000000000000000","in_use_wtc":"0.00000000","available_wtc":"0.00000000","xcoin_last_wtc":"273.8","total_cmt":"0.000000000000000000","in_use_cmt":"0.00000000","available_cmt":"0.00000000","xcoin_last_cmt":"0","total_theta":"0.000000000000000000","in_use_theta":"0.00000000","available_theta":"0.00000000","xcoin_last_theta":"86.73","total_poly":"0.000000000000000000","in_use_poly":"0.00000000","available_poly":"0.00000000","xcoin_last_poly":"0","total_loom":"0.000000000000000000","in_use_loom":"0.00000000","available_loom":"0.00000000","xcoin_last_loom":"16.55","total_cvc":"0.000000000000000000","in_use_cvc":"0.00000000","available_cvc":"0.00000000","xcoin_last_cvc":"0","total_man":"0.000000000000000000","in_use_man":"0.00000000","available_man":"0.00000000","xcoin_last_man":"0","total_waves":"0.00000000","in_use_waves":"0.00000000","available_waves":"0.00000000","xcoin_last_waves":"1002","total_itc":"0.000000000000000000","in_use_itc":"0.00000000","available_itc":"0.00000000","xcoin_last_itc":"106","total_true":"0.000000000000000000","in_use_true":"0.00000000","available_true":"0.00000000","xcoin_last_true":"317","total_abt":"0.000000000000000000","in_use_abt":"0.00000000","available_abt":"0.00000000","xcoin_last_abt":"0","total_link":"0.000000000000000000","in_use_link":"0.00000000","available_link":"0.00000000","xcoin_last_link":"2617","total_meetone":"0.00000000","in_use_meetone":"0.00000000","available_meetone":"0.00000000","xcoin_last_meetone":"0","total_horus":"0.00000000","in_use_horus":"0.00000000","available_horus":"0.00000000","xcoin_last_horus":"0","total_add":"0.00000000","in_use_add":"0.00000000","available_add":"0.00000000","xcoin_last_add":"0","total_salt":"0.00000000","in_use_salt":"0.00000000","available_salt":"0.00000000","xcoin_last_salt":"0","total_pst":"0.000000000000000000","in_use_pst":"0.00000000","available_pst":"0.00000000","xcoin_last_pst":"0","total_bhpc":"0.000000000000000000","in_use_bhpc":"0.00000000","available_bhpc":"0.00000000","xcoin_last_bhpc":"0","total_rnt":"0.000000000000000000","in_use_rnt":"0.00000000","available_rnt":"0.00000000","xcoin_last_rnt":"7.95","total_enj":"0.000000000000000000","in_use_enj":"0.00000000","available_enj":"0.00000000","xcoin_last_enj":"116.8","total_plx":"0.000000000","in_use_plx":"0.00000000","available_plx":"0.00000000","xcoin_last_plx":"1.419","total_vet":"0.000000000000000000","in_use_vet":"0.00000000","available_vet":"0.00000000","xcoin_last_vet":"3.598","total_mtl":"0.00000000","in_use_mtl":"0.00000000","available_mtl":"0.00000000","xcoin_last_mtl":"377.8","total_rdn":"0.000000000000000000","in_use_rdn":"0.00000000","available_rdn":"0.00000000","xcoin_last_rdn":"0","total_ins":"0.0000000000","in_use_ins":"0.00000000","available_ins":"0.00000000","xcoin_last_ins":"130.9","total_chl":"0.00000000","in_use_chl":"0.00000000","available_chl":"0.00000000","xcoin_last_chl":"0","total_black":"0.00000000","in_use_black":"0.00000000","available_black":"0.00000000","xcoin_last_black":"0","total_atd":"0.00000000","in_use_atd":"0.00000000","available_atd":"0.00000000","xcoin_last_atd":"0","total_iost":"0.00000000","in_use_iost":"0.00000000","available_iost":"0.00000000","xcoin_last_iost":"3.585","total_ocn":"0.000000000000000000","in_use_ocn":"0.00000000","available_ocn":"0.00000000","xcoin_last_ocn":"0","total_tmtg":"0.000000000000000000","in_use_tmtg":"0.00000000","available_tmtg":"0.00000000","xcoin_last_tmtg":"2.38","total_qkc":"0.000000000000000000","in_use_qkc":"0.00000000","available_qkc":"0.00000000","xcoin_last_qkc":"2.92","total_bznt":"0.000000000000000000","in_use_bznt":"0.00000000","available_bznt":"0.00000000","xcoin_last_bznt":"5.15","total_arn":"0.00000000","in_use_arn":"0.00000000","available_arn":"0.00000000","xcoin_last_arn":"0","total_hdac":"0.00000000","in_use_hdac":"0.00000000","available_hdac":"0.00000000","xcoin_last_hdac":"29.58","total_npxs":"0.000000000000000000","in_use_npxs":"0.00000000","available_npxs":"0.00000000","xcoin_last_npxs":"0.134","total_lba":"0.000000000000000000","in_use_lba":"0.00000000","available_lba":"0.00000000","xcoin_last_lba":"11.48","total_wet":"0.000000000000000000","in_use_wet":"0.00000000","available_wet":"0.00000000","xcoin_last_wet":"4.259","total_amo":"0.000000000000000000","in_use_amo":"0.00000000","available_amo":"0.00000000","xcoin_last_amo":"0.3198","total_bsv":"0.00000000","in_use_bsv":"0.00000000","available_bsv":"0.00000000","xcoin_last_bsv":"194100","total_keos":"0.00000000","in_use_keos":"0.00000000","available_keos":"0.00000000","xcoin_last_keos":"0","total_rom":"0.000000000000000000","in_use_rom":"0.00000000","available_rom":"0.00000000","xcoin_last_rom":"0","total_bxa":"0.000000000000000000","in_use_bxa":"0.00000000","available_bxa":"0.00000000","xcoin_last_bxa":"0","total_apis":"0.000000000000000000","in_use_apis":"0.00000000","available_apis":"0.00000000","xcoin_last_apis":"0.57","total_dac":"0.000000000000000000","in_use_dac":"0.00000000","available_dac":"0.00000000","xcoin_last_dac":"1.44","total_dacc":"0.00000000","in_use_dacc":"0.00000000","available_dacc":"0.00000000","xcoin_last_dacc":"0","total_auto":"0.000000000000000000","in_use_auto":"0.00000000","available_auto":"0.00000000","xcoin_last_auto":"0","total_orbs":"0.000000000000000000","in_use_orbs":"0.00000000","available_orbs":"0.00000000","xcoin_last_orbs":"10.98","total_tfuel":"0.000000000000000000","in_use_tfuel":"0.00000000","available_tfuel":"0.00000000","xcoin_last_tfuel":"0","total_valor":"0.000000000000000000","in_use_valor":"0.00000000","available_valor":"0.00000000","xcoin_last_valor":"252.4","total_con":"0.000000000000000000","in_use_con":"0.00000000","available_con":"0.00000000","xcoin_last_con":"2.672","total_ankr":"0.000000000000000000","in_use_ankr":"0.00000000","available_ankr":"0.00000000","xcoin_last_ankr":"1.736","total_mix":"0.000000000000000000","in_use_mix":"0.00000000","available_mix":"0.00000000","xcoin_last_mix":"1.769","total_vtho":"0.000000000000000000","in_use_vtho":"0.00000000","available_vtho":"0.00000000","xcoin_last_vtho":"0","total_lamb":"0.000000000000000000","in_use_lamb":"0.00000000","available_lamb":"0.00000000","xcoin_last_lamb":"14.55","total_cro":"0.00000000","in_use_cro":"0.00000000","available_cro":"0.00000000","xcoin_last_cro":"57.28","total_fx":"0.000000000000000000","in_use_fx":"0.00000000","available_fx":"0.00000000","xcoin_last_fx":"55.89","total_chr":"0.00000000","in_use_chr":"0.00000000","available_chr":"0.00000000","xcoin_last_chr":"15.2","total_mbl":"0.00000000","in_use_mbl":"0.00000000","available_mbl":"0.00000000","xcoin_last_mbl":"1.74","total_mxc":"0.000000000000000000","in_use_mxc":"0.00000000","available_mxc":"0.00000000","xcoin_last_mxc":"4.129","total_fab":"0.000000000000000000","in_use_fab":"0.00000000","available_fab":"0.00000000","xcoin_last_fab":"6.269","total_ogo":"0.000000000000000000","in_use_ogo":"0.00000000","available_ogo":"0.00000000","xcoin_last_ogo":"10.1","total_win":"0.00000000","in_use_win":"0.00000000","available_win":"0.00000000","xcoin_last_win":"0","total_dvp":"0.000000000000000000","in_use_dvp":"0.00000000","available_dvp":"0.00000000","xcoin_last_dvp":"3.839","total_fct":"0.000000000000000000","in_use_fct":"0.00000000","available_fct":"0.00000000","xcoin_last_fct":"21.7","total_fnb":"0.000000000000000000","in_use_fnb":"0.00000000","available_fnb":"0.00000000","xcoin_last_fnb":"1.422","total_fzz":"0.000000000000000000","in_use_fzz":"0.00000000","available_fzz":"0.00000000","xcoin_last_fzz":"1.84","total_trv":"0.000000000000000000","in_use_trv":"0.00000000","available_trv":"0.00000000","xcoin_last_trv":"9.883","total_pcm":"0.000000000000000000","in_use_pcm":"0.00000000","available_pcm":"0.00000000","xcoin_last_pcm":"3.172","total_dad":"0.000000000","in_use_dad":"0.00000000","available_dad":"0.00000000","xcoin_last_dad":"220","total_aoa":"0.000000000000000000","in_use_aoa":"0.00000000","available_aoa":"0.00000000","xcoin_last_aoa":"1.412","total_xsr":"0.000000000000000000","in_use_xsr":"0.00000000","available_xsr":"0.00000000","xcoin_last_xsr":"3.158","total_wom":"0.000000000000000000","in_use_wom":"0.00000000","available_wom":"0.00000000","xcoin_last_wom":"68.84","total_soc":"0.000000000000000000","in_use_soc":"0.00000000","available_soc":"0.00000000","xcoin_last_soc":"12.88","total_em":"0.00000000","in_use_em":"0.00000000","available_em":"0.00000000","xcoin_last_em":"3.495","total_qbz":"0.000000000000000000","in_use_qbz":"0.00000000","available_qbz":"0.00000000","xcoin_last_qbz":"10.31","total_boa":"0.0000000","in_use_boa":"0.00000000","available_boa":"0.00000000","xcoin_last_boa":"66.21","total_wpx":"0.000000000000000000","in_use_wpx":"0.00000000","available_wpx":"0.00000000","xcoin_last_wpx":"8.43","total_fleta":"0.000000000000000000","in_use_fleta":"0.00000000","available_fleta":"0.00000000","xcoin_last_fleta":"4.861","total_bnp":"0.000000000000000000","in_use_bnp":"0.00000000","available_bnp":"0.00000000","xcoin_last_bnp":"0.3455","total_sxp":"0.000000000000000000","in_use_sxp":"0.00000000","available_sxp":"0.00000000","xcoin_last_sxp":"470.8","total_cos":"0.000000000000000000","in_use_cos":"0.00000000","available_cos":"0.00000000","xcoin_last_cos":"5.77","total_el":"0.000000000000000000","in_use_el":"0.00000000","available_el":"0.00000000","xcoin_last_el":"1.569","total_hc":"0.00000000","in_use_hc":"0.00000000","available_hc":"0.00000000","xcoin_last_hc":"1162","total_bcd":"0.00000000","in_use_bcd":"0.00000000","available_bcd":"0.00000000","xcoin_last_bcd":"551.2","total_xvg":"0.00000000","in_use_xvg":"0.00000000","available_xvg":"0.00000000","xcoin_last_xvg":"3.022","total_xlm":"0.00000000","in_use_xlm":"0.00000000","available_xlm":"0.00000000","xcoin_last_xlm":"49","total_pivx":"0.00000000","in_use_pivx":"0.00000000","available_pivx":"0.00000000","xcoin_last_pivx":"282.2","total_etz":"0.000000000000000000","in_use_etz":"0.00000000","available_etz":"0.00000000","xcoin_last_etz":"0","total_gxc":"0.00000000","in_use_gxc":"0.00000000","available_gxc":"0.00000000","xcoin_last_gxc":"498.2","total_bhp":"0.00000000","in_use_bhp":"0.00000000","available_bhp":"0.00000000","xcoin_last_bhp":"748","total_btt":"0.00000000","in_use_btt":"0.00000000","available_btt":"0.00000000","xcoin_last_btt":"0.269","total_hyc":"0.000000000","in_use_hyc":"0.00000000","available_hyc":"0.00000000","xcoin_last_hyc":"1.351","total_vsys":"0.00000000","in_use_vsys":"0.00000000","available_vsys":"0.00000000","xcoin_last_vsys":"26.3","total_ipx":"0.000000000","in_use_ipx":"0.00000000","available_ipx":"0.00000000","xcoin_last_ipx":"102.5","total_wicc":"0.00000000","in_use_wicc":"0.00000000","available_wicc":"0.00000000","xcoin_last_wicc":"168.2","total_mbc":"0.00000000","in_use_mbc":"0.00000000","available_mbc":"0.00000000","xcoin_last_mbc":"0","total_luna":"0.00000000","in_use_luna":"0.00000000","available_luna":"0.00000000","xcoin_last_luna":"169.8","total_news":"0.00000000","in_use_news":"0.00000000","available_news":"0.00000000","xcoin_last_news":"0","total_aion":"0.000000000000000000","in_use_aion":"0.00000000","available_aion":"0.00000000","xcoin_last_aion":"71.7","total_kkkk":"0.000000000000000000","in_use_kkkk":"0.00000000","available_kkkk":"0.00000000","xcoin_last_kkkk":"0","total_cosm":"0.000000000000000000","in_use_cosm":"0.00000000","available_cosm":"0.00000000","xcoin_last_cosm":"10.52","total_mmm":"0.000000000","in_use_mmm":"0.00000000","available_mmm":"0.00000000","xcoin_last_mmm":"0"}}

**Comment by carloconnor** [comment]
The issue seems to be that the call to self.currency(code) returns a dictionary like this:
{'id': None, 'numericId': None, 'code': 'AMO', 'precision': None}

But the next line expects currency to be a string (like 'AMO' in this case)

**Comment by kroitor** [comment]
Will fix shortly.

**Comment by kroitor** [comment]
This was a minor bug introduced recently in #6694 . I've fixed it and it should work fine as of 1.25.42. We will be happy if you report back whether it is resolved on your side or not, when the build arrives. Thx for your involvement!

**Comment by carloconnor** [comment]
@kroitor I confirm it works with the new version, thanks again for such a quick fix!
